src/hash.h

```diff
@@ -89,7 +89,21 @@ inline uint256 Hash(const T1 p1begin, const T1 p1end,
     return result;
 }
 
-/** Compute the 160-bit hash an object. */
+/** Compute the 160-bit RIPEMD-160 hash of an array. */
+inline uint160 RipeMd160(const unsigned char* data, size_t size)
+{
+    uint160 result;
+    CRIPEMD160().Write(data, size).Finalize((unsigned char*)&result);
+    return result;
+}
+
+/** Compute the 160-bit RIPEMD-160 hash of a vector. */
+inline uint160 RipeMd160(const std::vector<unsigned char>& vch)
```

Comment:
Could make this a template function over things with `.data()` and `.size()` so that it works for `Span`s too. In which case you could also remove the other signature for `RipeMd160(ptr, size)` and just call `RipeMd160(Span<unsigned char>(ptr, size))`.

Reply:
I have a follow-up here: https://github.com/Empact/bitcoin/tree/base-blob-data that makes `RipeMd160` just a template and makes `base_blob` and descendants compatible by adding `base_blob::data()`. I figure it's best to put that up separately as it's mostly to do with `base_blob`.